hoverkite-protocol/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::ops::RangeInclusive;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Side {
+    Left,
+    Right,
+}
+
+impl Side {
+    pub fn opposite(self) -> Self {
+        match self {
+            Self::Left => Self::Right,
+            Self::Right => Self::Left,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Command {
+    // FIXME: stop using RangeInclusive, so we can derive Copy
+    SetMaxSpeed(RangeInclusive<i16>),
+    SetSpringConstant(u16),
+    SetTarget(i64),
+    Recenter,
+    BatteryReport,
+    Relax,
+    PowerOff,
+}
+
+#[cfg(feature = "std")]
+impl Command {
+    pub fn write_to(&self, mut writer: impl std::io::Write) -> std::io::Result<()> {
```

Comment:
(for the next PR): When looking at how to serialize on the hoverboard, I think I want to be able to `impl embedded_hal::serial::Write for std::io::Write` behind `#[cfg(feature = "std")]`, and then write everything in terms of `embedded_hal::serial::Write`. I think that this would need to happen in the embedded_hal crate though.

Reply:
I'm surprised that `std::io::Write` isn't part of `core`, really. It seems like it should be.

Reply:
Yeah, it's a bit silly that we have core::fmt::Write but not core::io::Write. I guess fmt is needed by webassembly things, whereas io is only needed by embedded?

In other news, it will end up being `impl<W: std::io::Write> embedded_hal::blocking::serial::Write<u8> for W`, because you need to be able to flush the whole thing without people dumping unrelated bytes in the middle of your packet. Luckily you already implemented the blocking trait.